File: src/domain/PDFSet.ts
export interface PDFSet {
  id: string
  name: string
  createdAt: string // ISO 8601 datetime string
  lastOpenedAt: string | null // ISO 8601 datetime string
  originalPdfPath: string
  translatedPdfPath: string
  bookmark: Bookmark | null
}

export interface Bookmark {
  page: number
  zoom: number
}

export interface CreatePDFSetParams {
  name: string
  originalPath: string
  translatedPath: string
}

File: src-tauri/src/models/pdf_set.rs
use chrono::{DateTime, Utc};
use serde::{Deserialize, Serialize};
use uuid::Uuid;

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct PDFSet {
    pub id: String,
    pub name: String,
    pub created_at: DateTime<Utc>,
    pub last_opened_at: Option<DateTime<Utc>>,
    pub original_pdf_path: String,
    pub translated_pdf_path: String,
    pub bookmark: Option<Bookmark>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct Bookmark {
    pub page: u32,
    pub zoom: f32,
}

impl PDFSet {
    pub fn new(name: String) -> Self {
        let id = Uuid::new_v4().to_string();
        let created_at = Utc::now();

        Self {
            id: id.clone(),
            name,
            created_at,
            last_opened_at: None,
            original_pdf_path: format!("pdfs/{}/original.pdf", id),
            translated_pdf_path: format!("pdfs/{}/translated.pdf", id),
            bookmark: None,
        }
    }
}
